src/backends/postgresql/session.cpp

```diff
@@ -29,6 +30,128 @@ void hard_exec(postgresql_session_backend & session_backend,
     postgresql_result(session_backend, PQexec(conn, query)).check_for_errors(errMsg);
 }
 
+// helper function to quote a string before sending to PostgreSQL
+std::string quote(PGconn * conn, std::string& s)
+{
+    int error_code;
+    std::string retv;
+    retv.resize(2 * s.length() + 3);
+    retv[0] = '\'';
+    size_t len_esc = PQescapeStringConn(conn, const_cast<char *>(retv.data()) + 1, s.c_str(), s.length(), &error_code);
+    if (error_code > 0)
+    {
+        len_esc = 0;
+    }
+    retv[len_esc + 1] = '\'';
+    retv.resize(len_esc + 2);
+
+    return retv;
+}
+
+// helper function to collect schemas from search_path
+std::vector<std::string> get_schema_names(postgresql_session_backend & session, PGconn * conn)
+{
+    std::vector<std::string> schema_names;
+    postgresql_result search_path_result(session, PQexec(conn, "SHOW search_path"));
+    if (search_path_result.check_for_data("search_path doesn't exist"))
+    {
+        if (PQntuples(search_path_result) > 0)
+        {
+            std::string search_path_content = PQgetvalue(search_path_result, 0, 0);
```

Comment:
Sorry, I hadn't realized it on first reading, but I think this code could be simplified like this:

```suggestion
        std::string search_path_content;
        if (PQntuples(search_path_result) > 0)
            search_path_content = PQgetvalue(search_path_result, 0, 0);
        if (search_path_content.empty())
            search_path_content = R"("$user", public)"; // fall back to default value
```

as then we could get rid of the `if` below.